modules/storages/app/common/storages/adapters/providers/share_point/queries/file_info_query.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Adapters
+    module Providers
+      module SharePoint
+        module Queries
+          class FileInfoQuery < Base
+            FIELDS = %w[id name fileSystemInfo file folder size createdBy lastModifiedBy parentReference webUrl].freeze
+
+            def call(auth_strategy:, input_data:)
+              identifier_hash = split_identifier(input_data.file_id)
+              drive_id = identifier_hash[:drive_id]
+              item_id = identifier_hash[:location]
+
+              make_base_query(auth_strategy:, drive_id:, item_id:).or do |error|
+                return Failure(error) unless error.code == :not_found && auth_strategy.value!.user.present?
+
+                admin_query(drive_id, item_id)
+              end
+            end
+
+            private
+
+            def make_base_query(auth_strategy:, drive_id:, item_id:)
```

Comment:
Trying to understand this code, I am wondering whether there is a better name for this method.

Firstly: It doesn't seem to make (i.e. build) a query, but it executes the query already. Additionally I am wondering _which_ query it is executing.

Would be also be fair to call this method `fetch_file_info` or `fetch_drive_item`?

Reply:
Similarly, there is the difference between `make_base_query` and `admin_query`. The latter seems to be the same as the former, just with different authentication.

By that logic `make_base_query` could also be called `user_query` (I don't like it, but it would be consistent) or `admin_query` could be renamed to something like `fetch_file_info_as_admin`.